### ISSUE TITLE
User-agent is not passing properly in some case

### ISSUE THREAD

**Issue by amkoro** [j0k3r/graby#162]
After j0k3r/safecurl package was updated to 2.0.3 it appears that User-Agent string is not passing properly while retrieving remote URL.
I'm using this code:
```
$options = [
    'extractor' => [
        'config_builder' => [
            'site_config' => ['/some/path'],
        ],
    ],
    'error_message' => 'my error message',
    'http_client' => [
        'timeout' => 60,
        'ua_browser' => 'my user agent string',
    ],
];

$defaults = [
    'cookies' => true,
    'allow_redirects' => [
        'max' => 3
    ],
];

$client = new Client(['handler' => new SafeCurlHandler(),
                      'defaults' => $defaults]);
$graby = new Graby($options, $client);
```

reverting j0k3r/safecurl to previous version solves the issue

**Comment by j0k3r** [comment]
Thanks for letting me know! I'll take a look

**Comment by j0k3r** [comment]
This looks strange because the only change to SafeCurl was to always add this header `Accept: */*`: https://github.com/j0k3r/safecurl/compare/2.0.2...master#diff-788df89fb53b1ab56b42871a2606b24b

And SafeCurl never take the user agent defined in the graby config.
It just validate the url is safe to be called.

Are you sure it's related to SafeCurl and not a change in graby for example?
Could you provide a better snippet on how I can reproduce the problem?

**Comment by amkoro** [comment]
yes, I'm completely sure that latest safecurl causing the issue
below are the Phiremock headers list for both cases:

"j0k3r/safecurl": "*"
```
[headers] => stdClass Object
                (
                    [Host] => Array
                        (
                            [0] => 127.0.0.1:8001
                        )

                    [Accept-Encoding] => Array
                        (
                            [0] => deflate, gzip
                        )

                    [Accept] => Array
                        (
                            [0] => */*
                        )

                )
```

"j0k3r/safecurl": "2.0.2"
```
[headers] => stdClass Object
                (
                    [Host] => Array
                        (
                            [0] => 127.0.0.1:8001
                        )

                    [User-Agent] => Array
                        (
                            [0] => Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.108 Safari/537.36
                        )

                    [Referer] => Array
                        (
                            [0] => http://www.google.co.uk/url?sa=t&source=web&cd=1
                        )

                )
```

**Comment by j0k3r** [comment]
Yeah I understand your problem. But I don't know how to reproduce.

I still need a snippet to reproduce the bug: a php snippet where you test that functionality isn't working anymore and how you check that.

Once I'll get that piece of code I'll be able to fix the problem.

**Comment by j0k3r** [comment]
I reverted the PR because we found other problem.
I'm still really interested in getting code from you to reproduce your bug so I might be able to fix it properly.

Is there any chance you can provide me the PHP code you are using to reproduce the bug?
Thanks :pray:

**Comment by j0k3r** [comment]
I released SafeCurl 2.0.4 to fix the problem.
